netbox/client.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2017 The go-netbox Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netbox
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+// A Client is a NetBox client.  It can be used to retrieve network and
+// datacenter infrastructure information from a NetBox server.
+type Client struct {
+	// DCIM provides access to methods in NetBox's DCIM API.
+	DCIM *DCIMService
+
+	// IPAM provides access to methods in NetBox's IPAM API.
+	IPAM *IPAMService
+
+	u      *url.URL
+	client *http.Client
+}
+
+// A API contains the Version of the netbox api (extracted from response
+// header) and a map[string]string of Endpoints
+type API struct {
+	Version   string
+	Endpoints map[string]string
+}
+
+type jsonErrorDetail struct {
```

Comment:
For the time being, let's make this an anonymous structure inside of `httpStatusOK`, since it's not used anywhere else.